README.md

```diff
@@ -0,0 +1,159 @@
+You can learn more about Triton backends in the [backend
+repo](https://github.com/triton-inference-server/backend). Ask
+questions or report problems on the [issues
+page](https://github.com/triton-inference-server/server/issues).
+This backend is designed to run
+[supported models](https://vllm.readthedocs.io/en/latest/models/supported_models.html)
+on a
+[vLLM engine](https://github.com/vllm-project/vllm/blob/main/vllm/engine/async_llm_engine.py).
+
+This is a Python-based backend. When using this backend, all requests are placed on the
+vLLM AsyncEngine as soon as they are received. Inflight batching and paged attention is handled
+by vLLM engine.
+
+Where can I ask general questions about Triton and Triton backends?
+Be sure to read all the information below as well as the [general
+Triton documentation](https://github.com/triton-inference-server/server#triton-inference-server)
+available in the main [server](https://github.com/triton-inference-server/server)
+repo. If you don't find your answer there you can ask questions on the
+main Triton [issues page](https://github.com/triton-inference-server/server/issues).
+
+## Building the vLLM Backend
+
+There are several ways to take advantage of the vLLM backend.
+
+### Option 1. Run the Docker Container.
+
+Pull the container with vLLM backend from [NGC](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/tritonserver) registry. This container has everything you need to run your vLLM model.
+
+### Option 2. Build a custom container from source
+You can follow steps described in the
+[Building With Docker](https://github.com/triton-inference-server/server/blob/main/docs/customization_guide/build.md#building-with-docker)
+guide and use the
+[build.py](https://github.com/triton-inference-server/server/blob/main/build.py)
+script.
+
+A sample command to build a Triton Server container with all options enabled is shown below. Feel free to customize flags according to your needs.
+
+```
+./build.py -v  --enable-logging --enable-stats --enable-tracing
+                --enable-metrics --enable-gpu-metrics --enable-cpu-metrics
+                --enable-gpu
+                --filesystem=gcs --filesystem=s3 --filesystem=azure_storage
+                --endpoint=http --endpoint=grpc --endpoint=sagemaker --endpoint=vertex-ai
+                --upstream-container-version=23.10
+                --backend=python:r23.10
+                --backend=vllm:r23.10
+```
+
+### Option 3. Add the vLLM Backend to the Triton Container
+
+You can install the vLLM backend directly into the NGC Triton container.
+In this case, please install vLLM first. You can do so by running
+`pip install vllm==<vLLM_version>`. Then, set up the vLLM backend in the
+container with the following commands:
+
+```
+mkdir -p /opt/tritonserver/backends/vllm
+wget -P /opt/tritonserver/backends/vllm https://raw.githubusercontent.com/triton-inference-server/vllm_backend/main/src/model.py
+```
```

Comment:
Not an action item here, but a random food for thought that could be nice for both users and developers. If we standardize on a certain python-based-backend git repository structure, we can do something like:
```
git clone https://github.com/triton-inference-server/vllm_backend.git /opt/tritonserver/backends
```

1. Single command
2. Developers could iterate on the backend directly and just reload triton without copying files/builds around
3. More support for multi-file implementations. The `wget` is nice, but won't scale past a single file. Ex: Imagine `model.py` implements `TritonPythonModel` but imports `implementation.py` that has all the gorey details for certain features.

Just some random Tuesday ideas in my head. Core would just be updated to also look for `src/model.py` or something instead of just `model.py`.

Reply:
this will not work with git clone, since required `model.py` is in sub-directory of `vllm_backend`, plus `clone` will clone tests as well.

We can discuss the best solution at some point.

Reply:
For the ease of development, I think your earlier idea of symlinks makes more sense.

Reply:
We have a separate goal of improving python backend developer experience (more for things like debugging, ipdb, etc) somewhere in the pipeline, so this came to mind as a tangential idea.

Reply:
I see, by any chance, do you know in what ticket this is tracked? If you don't remember, then no worries